### ISSUE TITLE
Semantic Sub-Effecting Problems on Base Effects

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#8773]
The current base effects are:

```flix
eff Exec
eff FileRead
eff FileWrite
eff IO
eff Net
eff NonDet
eff Sys
eff Time
```

Problem: The effects are not entirely disjoint which suggests that they have a semantic sub-effecting relationship.

For example:

- We can get `Time` with `FileRead` by doing `File.read(f); File.accessTime(f)`. Does this mean that `Time` is a super-effect of `FileRead`?
- We can get `FileRead` with `Exec` by doing `Exec.exec("cat foo.txt"); ...`. Does this mean that `FileRead` is a super-effect of `Exec`?
- In some sense all the effects are `NonDet`. 
- We can get any effect with `Sys` using reflection.

**Comment by magnus-madsen** [comment]
@mlutze @JonathanStarup What are the implications of this analysis? 

Does it mean we should collapse all effects into just `IO`?

Does it mean that there is an implicit sub-effecting relationship, but the compiler is just not smart enough to see it? (Perhaps a bit in the same way that there are two different Java classes: `Path` and `File` which seem related but are not according to the typing relation.)

Does it mean that `File.read` should have both `FileRead` and `Time`? 

What would you do here?

**Comment by JonathanStarup** [comment]
I also had the same thought for a solution, if we are gonna have it.

**Comment by magnus-madsen** [comment]
Is that workable? It seems like a solution to an exercise, but actually programming with it would seem confusing and annoying? Or?

**Comment by JonathanStarup** [comment]
Yeah it essentially means that all file functions have `File + Time`

With a very manual approach you could also just have `File.accessTime(f): _ \ File + Time`

**Comment by magnus-madsen** [comment]
> Yeah it essentially means that all file functions have `File + Time`
> 
> With a very manual approach you could also just have `File.accessTime(f): _ \ File + Time`

This does not spark joy.

**Comment by magnus-madsen** [comment]
We dropped primitive effects (for now).
